### PR TITLE
Don't depend on the `moment` prop from `localize`.

### PR DESCRIPTION
**Pull request by sgomes**
The `localize` HOC from `i18n-calypso` adds a `moment` prop to wrapped components.

If we're to remove the `moment` dependency from `i18n-calypso`, we'll need to stop components depending on this prop, and instead using the same prop from the `withLocalizedMoment` HOC.

This PR attempts to change every instance of a dependency on `localize`'s `moment` prop into a dependency on `withLocalizedMoment`'s `moment` prop.

**Note**: This PR will report a number of linting issues. These were already present, and I decided it was out of scope to try to fix them, particularly taking into account the scale of the PR and the amount of work needed to get things cleaned up in some of these files.

#### Changes proposed in this Pull Request

* Add `withLocalizedMoment` HOC to every component depending on a `moment` prop.
* Make `apps/notifications` check the locale slug from `i18n-calypso` instead of using the `withLocalizedMoment` HOC. We can't make `apps/notifications` depend on `localized-moment` without extracting that out, which would be complicated due to how it's tied to state in Calypso. The user block notifications will fail to update timestamps when the user changes locales, but that's already the current behaviour.
* Minor drive-by fixes on changed files.

#### Testing instructions

This is one of those large PRs that is hard to test for. I'd recommend a thorough code inspection (there are many files, but the changes are mostly mechanical), and the rest I'd leave to the E2E suite.

### CODE REVIEW

**Comment by sgomes** [comment]
The E2E errors seem to be consistent, so they may be related to the PR. Investigating.

**Comment by tyxla** [comment]
This is great work 🙌 

I couldn't review this until now and will be AFK for the remaining part of the week. Sorry about that, I'm happy to review next week if y'all haven't merged it by then.

**Comment by sgomes** [comment]
> This is great work 🙌
> 
> I couldn't review this until now and will be AFK for the remaining part of the week. Sorry about that, I'm happy to review next week if y'all haven't merged it by then.

Thanks, @tyxla! 🙂No worries, this can definitely wait until then. Enjoy your AFK!

**Comment by sgomes** [comment]
Rebased on master after `prettier` changes.

**Comment by sgomes** [comment]
Rebased on master after `@format` pragma changes.

**Review by tyxla — COMMENTED** [review]
Really nice work 👏 

I spent some time smoke-testing it and didn't find any problems. Nice!

I also took a thorough look at the code, and while it looks great, there are some more occurrences of either direct `moment` import or a `i18n-calypso` one in:

* `client/components/post-schedule/index.jsx`
* `client/my-sites/checkout/checkout-thank-you/index.jsx`
* `client/my-sites/plugins/plugin-meta/index.jsx`

should we use `withLocalizedMoment` there too? Or are they skipped on purpose?

**Comment by sgomes** [comment]
> Really nice work 👏
> 
> I spent some time smoke-testing it and didn't find any problems. Nice!
> 
> I also took a thorough look at the code, and while it looks great, there are some more occurrences of either direct `moment` import or a `i18n-calypso` one in:
> 
> * `client/components/post-schedule/index.jsx`
> * `client/my-sites/checkout/checkout-thank-you/index.jsx`
> * `client/my-sites/plugins/plugin-meta/index.jsx`
> 
> should we use `withLocalizedMoment` there too? Or are they skipped on purpose?

Oh, there's plenty of work needed beyond this PR, definitely :) I have a bunch of PRs open on the same general issue.

This PR only deals with one specific type of change that's needed: components that are using the `moment` prop from the `localize` HOC. Other PRs that I have open deal separately with modules that import the `moment` export from `i18n-calypso`, for example.

File: apps/notifications/src/panel/templates/block-user.jsx [hunk 1]
```diff
@@ -1,10 +1,16 @@
+/**
+ * External dependencies
+ */
 import React from 'react';
 import { connect } from 'react-redux';
-import { localize } from 'i18n-calypso';
+import { localize, getLocaleSlug } from 'i18n-calypso';
+import moment from 'moment';
```
[the remark this answers is not in the document]
Comment (sgomes) [reply apps/notifications/src/panel/templates/block-user.jsx]: That's right, `apps` doesn't have access to `components/localized-moment` without some serious changes. This simpler change allows us to drop the dependency on `localize`'s `moment` without rearchitecting the whole build process.

**Comment by tyxla** [comment]
> This PR only deals with one specific type of change that's needed: components that are using the moment prop from the localize HOC. Other PRs that I have open deal separately with modules that import the moment export from i18n-calypso, for example.

Sounds great! I spent over 30 minutes smoke testing this, and I couldn't break it. I'd say it's ready to go after a rebase.

File: apps/notifications/src/panel/templates/block-user.jsx [hunk 2]
```diff
@@ -39,7 +45,8 @@ export class UserBlock extends React.Component {
 			return '';
 		}
 
-		momentTime = this.props.moment( timestamp );
+		const localeSlug = getLocaleSlug();
```
[the remark this answers is not in the document]
Comment (sgomes) [reply apps/notifications/src/panel/templates/block-user.jsx]: I think that's more of a side-effect of the history of the code and how it was pulled into the monorepo than any sort of design decision. This file is still subject to the same lint rules as everything else (and it's all a big blob of red underlines, as a consequence), and it goes through the same build process, so I think it's just that no-one's had the opportunity to go in and modernise it yet. I could make this a `var`, but that would just be one more thing to fix when that work eventually happens :)

[the remark this answers is not in the document]
Comment (tyxla) [reply apps/notifications/src/panel/templates/block-user.jsx]: Makes sense, thanks 😉 Let's keep it as-is.

**Comment by sgomes** [comment]
> > This PR only deals with one specific type of change that's needed: components that are using the moment prop from the localize HOC. Other PRs that I have open deal separately with modules that import the moment export from i18n-calypso, for example.
> 
> Sounds great! I spent over 30 minutes smoke testing this, and I couldn't break it. I'd say it's ready to go after a rebase.

Thank you for all your testing! I'll get started on that rebase.

**Comment by sgomes** [comment]
Rebase complete. As expected, only the `lint` check failed.

**Review by tyxla — APPROVED** [review]
LGTM, thank you! 🚢

**Pull request merged**